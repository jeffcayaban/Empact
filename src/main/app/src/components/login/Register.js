import React from 'react'
import { Button, Col, ControlLabel, FormControl, FormGroup, Glyphicon, Grid, HelpBlock, OverlayTrigger, Panel,
    Popover} from "react-bootstrap";
import {checkUsernameAvailability, signup} from "../../utils/AppUtils";
import swal from "sweetalert2";
import {
    generateErrorDialog,
    generateSuccessDialog,
    sendLoginRequest
} from "../../utils/Helpers";
import RegisterFormField from "./RegisterFormField";
import {
    CONFIRM_PASSWORD_ID,
    FIRST_NAME_ID, FIRST_NAME_LENGTH_MAX,
    LAST_NAME_ID, LAST_NAME_LENGTH_MAX,
    PASSWORD_ID, PASSWORD_LENGTH_MAX,
    PASSWORD_LENGTH_MIN,
    USERNAME_ID, USERNAME_LENGTH_MAX
} from "../../constants";

/**
 * Is the page used for users to create a new user account.
 */

class Register extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formFields: {},
            formValidationStates: {},

            isUsernameChosen: false,
            isCreating: false,
            isCheckingUsername: false,
            usernameLength: 0
        };
    }

    // Sends the request to log in into an existing account.
    handleSignIn() {
        const loginRequest = {};
        loginRequest[USERNAME_ID] = this.state.formFields[USERNAME_ID];
        loginRequest[PASSWORD_ID] = this.state.formFields[PASSWORD_ID];
        sendLoginRequest(loginRequest, true, this.state.formFields[USERNAME_ID], null);
    }

    // Confirms the request to create a new user account.
    onSubmit(newUserData) {
        signup(newUserData)
            .then(response => {
                generateSuccessDialog('Your account has been created!', 'Okay').then((result) => {
                    // Log into the newly created account.
                    (result.value) && this.handleSignIn();
                })
            }).catch(error => generateErrorDialog('There was a problem registering your account'))
    }

    // Sends the request to create a new user account.
    handleSubmit(e) {
        e.preventDefault();
        const formFields = this.state.formFields;

        if (formFields[PASSWORD_ID] !== formFields[CONFIRM_PASSWORD_ID]) {
            generateErrorDialog('Both passwords do not match. Please try again.');

        } else {
            this.setState({ isCreating: true });

            const registerRequest = {};

            registerRequest[USERNAME_ID] = formFields[USERNAME_ID];
            registerRequest[FIRST_NAME_ID] = formFields[FIRST_NAME_ID];
            registerRequest[LAST_NAME_ID] = formFields[LAST_NAME_ID];
            registerRequest[PASSWORD_ID] = formFields[PASSWORD_ID];

            this.setState({ isCreating: false });
            this.onSubmit(registerRequest);
        }
    }

    // Updates a new user account field value within the component's state.
    handleChange(formField, event) {
        const { formFields, formValidationStates } = this.state;
        formFields[formField] = event.target.value;
        formValidationStates[formField] = (event.target.value.length <= 0) ? "error" : null;
        this.setState({ formFields, formValidationStates });
    }

    // Updates the component's state to indicate whether a username is chosen.
    onChangeUsername() {
        this.setState({ isUsernameChosen: false });
    }

    // Checks whether a given username is available to be taken.
    checkUsernameAvailability(e) {
        e.preventDefault();
        this.setState({ isCheckingUsername: true });

        checkUsernameAvailability(this.state.formFields[USERNAME_ID])
            .then(response => {
                if (!response.usernameExists) {
                    // Assumes the username is available.

                    generateSuccessDialog(
                        'Click on continue to start filling out the required details for your account.',
                        'Continue'
                    ).then(() => {
                        this.setState({ isUsernameChosen: true, isCheckingUsername: false });
                    });
                } else {
                    // Assumes the username is not available.

                    swal({
                        title: 'Username is not available',
                        text: 'Please use a different username',
                        type: 'error',
                    });

                    this.setState({ isCheckingUsername: false });
                }
            })
            .catch(error => {
                // Assumes that the server has failed to check the username's availability.

                generateErrorDialog('There was a problem checking the availability of this username.');
                this.setState({ isCheckingUsername: false })
            })
    }

    // Generates a popover to inform the user of restrictions on the password.
    static generatePasswordPopover() {
        return <Popover id={"password-info-popover"}>Your password must be between 5 and 21 characters.</Popover>;
    }

    // Generates a form for the user to enter the details of the new account.
    generateUserDetailForm() {
        const { formValidationStates, formFields, isCreating } = this.state;

        return (
            <Panel>
                <Panel.Heading>
                    <Panel.Title componentClass="h3">2. Fill out your details</Panel.Title>
                </Panel.Heading>
                <Panel.Body>
                    <form onSubmit={this.handleSubmit.bind(this)}>
                        <FormGroup controlId={USERNAME_ID}>
                            <ControlLabel>Username</ControlLabel>
                            <FormControl type={"text"} value={formFields[USERNAME_ID]} disabled />
                        </FormGroup>
                        <RegisterFormField
                            id={FIRST_NAME_ID}
                            validationState={formValidationStates[FIRST_NAME_ID]}
                            handleChange={this.handleChange.bind(this, FIRST_NAME_ID)}
                            name={"First Name"}
                            type={"text"}
                            placeholder={"Enter your first name"}
                            maxLength={FIRST_NAME_LENGTH_MAX}
                        />
                        <RegisterFormField
                            id={LAST_NAME_ID}
                            validationState={formValidationStates[LAST_NAME_ID]}
                            handleChange={this.handleChange.bind(this, LAST_NAME_ID)}
                            name={"Last Name"}
                            type={"text"}
                            placeholder={"Enter your last name"}
                            maxLength={LAST_NAME_LENGTH_MAX}
                        />
                        <FormGroup controlId={PASSWORD_ID} validationState={formValidationStates[PASSWORD_ID]}>
                            <span>
                                <ControlLabel>Password</ControlLabel>
                                <OverlayTrigger trigger={['hover', 'focus']} placement="top"
                                                overlay={Register.generatePasswordPopover()}>
                                  <Glyphicon glyph="info-sign" style={{ paddingLeft: '5px' }}/>
                                </OverlayTrigger>
                            </span>
                            <FormControl
                                type={"password"}
                                placeholder={"Enter a password"}
                                onChange={this.handleChange.bind(this, PASSWORD_ID)}
                                required={true}
                                minLength={PASSWORD_LENGTH_MIN}
                                maxLength={PASSWORD_LENGTH_MAX}
                            />
                        </FormGroup>
                        <RegisterFormField
                            id={CONFIRM_PASSWORD_ID}
                            validationState={formValidationStates[CONFIRM_PASSWORD_ID]}
                            handleChange={this.handleChange.bind(this, CONFIRM_PASSWORD_ID)}
                            name={"Confirm Password"}
                            type={"password"}
                            placeholder={"Enter your password again"}
                            minLength={PASSWORD_LENGTH_MIN}
                            maxLength={PASSWORD_LENGTH_MAX}
                        />
                        <div className={"buttonContainer"}>
                            <div style={{ paddingRight: '10px' }}>
                                <Button onClick={this.onChangeUsername.bind(this)}>Change Username</Button>
                            </div>
                            <Button bsStyle="primary" type="submit" disabled={isCreating}>
                                { isCreating ? 'Registering...' : 'Register' }
                            </Button>
                        </div>
                    </form>
                </Panel.Body>
            </Panel>
        )
    }

    render() {
        const { formValidationStates, isUsernameChosen, usernameLength, isCheckingUsername } = this.state;

        return (
            <div>
                <div className={"centerStyle"}>
                    <h2 align="center">Register</h2>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"} style={{ paddingTop: '30px' }}>
                        <Col md={12} lg={12}>
                            <Panel>
                                <Panel.Heading>
                                    <Panel.Title componentClass="h3">1. Check if your username is available</Panel.Title>
                                </Panel.Heading>
                                <Panel.Body>
                                    <form onSubmit={this.checkUsernameAvailability.bind(this)}>
                                        <FormGroup validationState={formValidationStates[USERNAME_ID]}>
                                            <ControlLabel id={"username-label"}>Username</ControlLabel>
                                            <FormControl
                                                id={"username-input"}
                                                type={"text"}
                                                placeholder={"Enter a Username"}
                                                onChange={(e) => {
                                                    this.handleChange(USERNAME_ID, e);
                                                    this.setState({ usernameLength: e.target.value.length })
                                                }}
                                                required={true}
                                                maxLength={USERNAME_LENGTH_MAX}
                                                disabled={isUsernameChosen}
                                            />
                                            <FormControl.Feedback />
                                            <div id={"usernameFieldValidation"}>
                                                <div id={"usernameFieldError"}>
                                                    { formValidationStates[USERNAME_ID] === 'error' &&
                                                    <HelpBlock>This field cannot be empty.</HelpBlock> }
                                                </div>
                                                <div style={{ resizeMode: 'contain', paddingTop: '5px' }}>
                                                    <p>{usernameLength}/15</p>
                                                </div>
                                            </div>
                                        </FormGroup>
                                        <Button bsStyle="primary" type="submit"
                                                disabled={isCheckingUsername || isUsernameChosen}>
                                            { isCheckingUsername ? 'Checking...' : 'Check Availability' }
                                        </Button>
                                    </form>
                                </Panel.Body>
                            </Panel>
                            { isUsernameChosen && this.generateUserDetailForm() }
                        </Col>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default Register;