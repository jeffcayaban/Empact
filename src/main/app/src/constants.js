export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;