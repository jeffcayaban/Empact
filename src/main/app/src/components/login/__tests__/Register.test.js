import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import * as AppUtils from '../../../utils/AppUtils';
import * as Helpers from '../../../utils/Helpers';

import Register from "../Register";
import {PASSWORD_ID} from "../../../constants";

describe('Register', () => {

    it('should render successfully', () => {
        const component = shallow(<Register />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully call generateUserDetailForm', () => {
        const component = shallow(<Register />);
        const instance = component.instance();
        instance.generateUserDetailForm();

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully call checkUsernameAvailability if the username is available', () => {
        const originalCheckUsernameAvailability = AppUtils.checkUsernameAvailability;
        AppUtils.checkUsernameAvailability = () => Promise.resolve({ available: true });

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.checkUsernameAvailability({ preventDefault: jest.fn() });

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.checkUsernameAvailability = originalCheckUsernameAvailability;
    });

    it('should successfully call checkUsernameAvailability if the username is unavailable', () => {
        const originalCheckUsernameAvailability = AppUtils.checkUsernameAvailability;
        AppUtils.checkUsernameAvailability = () => Promise.resolve({ available: false });

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.checkUsernameAvailability({ preventDefault: jest.fn() });

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.checkUsernameAvailability = originalCheckUsernameAvailability;
    });

    it('should successfully call checkUsernameAvailability if the api fails', () => {
        const originalCheckUsernameAvailability = AppUtils.checkUsernameAvailability;
        AppUtils.checkUsernameAvailability = () => Promise.reject();

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.checkUsernameAvailability({ preventDefault: jest.fn() });

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.checkUsernameAvailability = originalCheckUsernameAvailability;
    });

    it('should successfully call handleSignIn', () => {
        const originalSendLoginRequest = Helpers.sendLoginRequest;
        Helpers.sendLoginRequest = jest.fn();

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.handleSignIn();

        expect(Helpers.sendLoginRequest).toBeCalled();

        Helpers.sendLoginRequest = originalSendLoginRequest;
    });

    it('should successfully call onSubmit', () => {
        const originalSignup = AppUtils.signup;
        AppUtils.signup = () => Promise.resolve({ value: true });

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.onSubmit("testData");

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.signup = originalSignup;
    });

    it('should generate an error if signup fails', () => {
        const originalSignup = AppUtils.signup;
        AppUtils.signup = () => Promise.reject();

        const component = shallow(<Register />);
        const instance = component.instance();
        instance.onSubmit("testData");

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.signup = originalSignup;
    });

    it('should successfully call handleSubmit', () => {
        const component = shallow(<Register />);
        component.setState({ formFields: {
                PASSWORD_ID: 'test',
                CONFIRM_PASSWORD_ID: 'test'
            }});

        const instance = component.instance();

        instance.handleSubmit({ preventDefault: jest.fn() });

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully call handleChange', () => {
        const component = shallow(<Register />);
        component.setState({ formFields: { PASSWORD_ID: 'test' }, formValidationStates: { PASSWORD_ID: 'test'}});

        const instance = component.instance();
        instance.handleChange(PASSWORD_ID, { target: { value: 'testValue' }});

        expect(component.state('formFields')[PASSWORD_ID]).toBe('testValue');
    });

    it('should successfully call onChangeUsername', () => {
        const component = shallow(<Register />);
        const instance = component.instance();
        instance.onChangeUsername();
        expect(component.state('isUsernameChosen')).toBe(false);
    });
});
