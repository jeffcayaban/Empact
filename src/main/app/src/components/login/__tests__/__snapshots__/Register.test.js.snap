// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should generate an error if signup fails 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                type="submit"
              >
                Check Availability
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should render successfully 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                type="submit"
              >
                Check Availability
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call checkUsernameAvailability if the api fails 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={true}
                type="submit"
              >
                Checking...
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call checkUsernameAvailability if the username is available 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={true}
                type="submit"
              >
                Checking...
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call checkUsernameAvailability if the username is unavailable 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={true}
                type="submit"
              >
                Checking...
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call generateUserDetailForm 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                type="submit"
              >
                Check Availability
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call handleSubmit 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                type="submit"
              >
                Check Availability
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;

exports[`Register should successfully call onSubmit 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      Register
    </h2>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Panel)>
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              1. Check if your username is available
            </PanelTitle>
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  id="username-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  disabled={false}
                  id="username-input"
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="Enter a Username"
                  required={true}
                  type="text"
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
                <div
                  id="usernameFieldValidation"
                >
                  <div
                    id="usernameFieldError"
                  />
                  <div
                    style={
                      Object {
                        "paddingTop": "5px",
                        "resizeMode": "contain",
                      }
                    }
                  >
                    <p>
                      0
                      /15
                    </p>
                  </div>
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                type="submit"
              >
                Check Availability
              </Button>
            </form>
          </PanelBody>
        </Uncontrolled(Panel)>
      </Col>
    </Grid>
  </div>
</div>
`;
